packages/backend/src/nest/connections-manager/connections-manager.service.ts

```diff
@@ -286,20 +282,30 @@ export class ConnectionsManagerService extends EventEmitter implements OnModuleI
     await this.socketService.init()
   }
 
-  public async leaveCommunity(): Promise<boolean> {
+  public async leaveCommunity() {
     this.logger('Running leaveCommunity')
 
-    this.logger('Resetting tor')
-    this.tor.resetHiddenServices()
+    await this.closeSocket()
+
+    if (this.storageService) {
+      this.logger('Stopping OrbitDB')
+      await this.storageService.stopOrbitDb()
+    }
 
-    this.logger('Closing the socket')
-    this.closeSocket()
+    if (this.libp2pService) {
+      this.logger('Stopping libp2p')
+      await this.libp2pService.close()
+    }
 
-    this.logger('Purging local DB')
-    await this.localDbService.purge()
+    if (this.localDbService) {
+      this.logger('Purging local DB')
+      await this.localDbService.purge()
+      this.logger('Closing local DB')
+      await this.localDbService.close()
+    }
 
-    this.logger('Closing services')
-    await this.closeAllServices({ saveTor: true })
+    this.logger('Resetting tor')
+    this.tor.resetHiddenServices()
 
     this.logger('Purging data')
     await this.purgeData()
```

Comment:
The important piece here is purging/closing LevelDB after stopping OrbitDB. However, I didn't want to get into a larger refactor so in order to make that work, I copied the contents of `closeAllServices` into this function. Other things still depend on close all services. I think the only difference between the two is that this function purges LevelDB and `closeAllServices` doesn't. So we could add another parameter to `closeAllServices` like `purgeLocalDb: boolean` or something.